src/ComputeSharp/Graphics/Resources/Abstract/Texture2D{T}.cs

```diff
@@ -147,22 +147,22 @@ private protected Texture2D(GraphicsDevice device, int width, int height, Resour
     /// </summary>
     /// <param name="destination">The target memory area to write data to.</param>
     /// <param name="size">The size of the target memory area to write data to.</param>
-    /// <param name="x">The horizontal offset in the source texture.</param>
-    /// <param name="y">The vertical offset in the source texture.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
     /// <param name="width">The width of the memory area to copy.</param>
     /// <param name="height">The height of the memory area to copy.</param>
-    internal void CopyTo(ref T destination, int size, int x, int y, int width, int height)
+    internal void CopyTo(ref T destination, int size, int sourceOffsetX, int sourceOffsetY, int width, int height)
     {
         GraphicsDevice.ThrowIfDisposed();
 
         ThrowIfDisposed();
 
-        Guard.IsInRange(x, 0, Width, nameof(x));
-        Guard.IsInRange(y, 0, Height, nameof(y));
+        Guard.IsInRange(sourceOffsetX, 0, Width, nameof(sourceOffsetX));
+        Guard.IsInRange(sourceOffsetY, 0, Height, nameof(sourceOffsetY));
         Guard.IsBetweenOrEqualTo(width, 1, Width, nameof(width));
         Guard.IsBetweenOrEqualTo(height, 1, Height, nameof(height));
```

Comment:
Aren't these checked by the checking the offset is positive and offset+size is in bounds check below?

Reply:
See answer above.